pubsub/api/Program.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace PubSubSample
+{
+
+  public class Program
+  {
+    public static void Main(string[] args)
+    {
+      if (args.Length == 0)
+        PrintUsage();
+      else
+        RunCommand(args.First(), args.Skip(1).ToArray());
+    }
+
+    static void PrintUsage()
+    {
+      Console.WriteLine(@"Usage: PubSubSample.exe [command] [args]

```

Comment:
Samples are separated into files so that each sample snippet can include `using` statements.

See existing Java & Python samples, eg. [Pub/Sub Subscriber sample](https://cloud.google.com/pubsub/subscriber#receive_pull)

It is useful for the `using` statements that are required for each snippet to appear alongside that snippet.  This is also useful because, if the required `using` statements are not correct:
 1. The project will fail to build (*missing statements*)
 1. Visual Studio will show you unused statements (*unnecessary statements*)